pkg/collector/corechecks/oracle-dbm/statements.go

```diff
@@ -432,16 +440,55 @@ func (c *Check) StatementMetrics() (int, error) {
 	var oracleRows []OracleRow
 	var planErrors uint16
 	if c.config.QueryMetrics.Enabled {
-		statementMetrics, err := GetStatementsMetricsForKeys(c, "force_matching_signature", "!")
+		var statementMetrics []StatementMetricsDB
+		var SQLID string
+		if c.config.QueryMetrics.DisableLastActive {
+			SQLID = "max(sql_id)"
+		} else {
+			SQLID = "sq.sql_id"
+		}
+		sql := fmt.Sprintf("%s %s as sql_id, %s %s", SELECT_FMS, SQLID, getColumnSums(), FROM)
+		if !c.config.QueryMetrics.DisableLastActive {
+			sql = fmt.Sprintf("%s, %s", sql, LAST_ACTIVE_FMS)
+		}
+		sql = fmt.Sprintf("%s %s", sql, WHERE)
+		if !c.config.QueryMetrics.DisableLastActive {
+			sql = fmt.Sprintf("%s AND sq.force_matching_signature = s.force_matching_signature", sql)
+		}
+		sql = fmt.Sprintf("%s %s", sql, GROUP_BY_FMS)
+		if !c.config.QueryMetrics.DisableLastActive {
+			sql = fmt.Sprintf("%s, %s", sql, SQLID)
+		} else {
+			sql = fmt.Sprintf("%s %s", sql, "HAVING MAX ( last_active_time ) > sysdate - :seconds/24/60/60")
+		}
+		sql = fmt.Sprintf("%s %s", sql, FETCH)
```

Comment:
I changed it. The only part that is factored out are the columns - it's a large, repetitive text, so I thought it would be better to have it so. But if you think, it would be better to have it embedded in the query text, let me know, and I'll change it.

Reply:
I'm actually on the fence about this one. I really like having the ability to copy paste a sql query from the code with the fewest modifications/replacement needed (to offer an updated query for a customer to run when troubleshooting, for example) but I agree there's a lot of repetition here. I'll let you decide what you think would make the more sense here. We can revisit later.

Reply:
perhaps we could just document the queries internally in our support FAQ? that would actually be really nice to have for the other integrations too

Reply:
Yes, I think the maintenance of that documentation would be the tricky part. We had discussed at some point having built-in functionality to generate the queries that the agent runs along with the metrics / data associated with those. It seems a little like a feature for the future but 🤷 .

Reply:
>I think the maintenance of that documentation would be the tricky part

yes, that's a good point

Reply:
I denormalised the columns names.